src/gc/tile_group_compactor.cpp

```diff
@@ -0,0 +1,194 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// transaction_level_gc_manager.cpp
+//
+// Identification: src/gc/transaction_level_gc_manager.cpp
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "gc/tile_group_compactor.h"
+
+
+namespace peloton {
+namespace gc {
+
+void TileGroupCompactor::CompactTileGroup(const oid_t &tile_group_id) {
+
+  size_t attempts = 0;
+  size_t max_attempts = 100;
+
+  constexpr auto kMinPauseTime = std::chrono::microseconds(1);
+  constexpr auto kMaxPauseTime = std::chrono::microseconds(100000);
+
+  auto pause_time = kMinPauseTime;
+
+  while (attempts < max_attempts) {
+
+    auto tile_group = catalog::Manager::GetInstance().GetTileGroup(tile_group_id);
+    if (tile_group == nullptr) {
+      return; // this tile group no longer exists
+    }
+
+    storage::DataTable *table =
+        dynamic_cast<storage::DataTable *>(tile_group->GetAbstractTable());
+
+    if (table == nullptr) {
+      return; // this table no longer exists
+    }
+
+    bool success = MoveTuplesOutOfTileGroup(table, tile_group);
+
```

Comment:
I find there is only one usage of function MoveTuplesOutOfTileGroup() and it is here. It seems that the tuples in the given tile group is always moved to the tile group's table. Then I think there may not be necessary to have a data table as argument in MoveTuplesOutOfTileGroup(), since we know that this table is this tile group's table.